lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_firebase_test.rb

```diff
@@ -0,0 +1,168 @@
+require 'securerandom'
+
+module Fastlane
+  module Actions
+    module SharedValues
+      FIREBASE_TEST_RESULT = :FIREBASE_TEST_LOG_FILE
+      FIREBASE_TEST_LOG_FILE_PATH = :FIREBASE_TEST_LOG_FILE_PATH
+    end
+
+    class AndroidFirebaseTestAction < Action
+      def self.run(params)
+        # Log in to Firebase (and validate credentials)
+        test_runner = Fastlane::FirebaseTestRunner.new(key_file: params[:key_file])
+        run_uuid = params.fetch(:test_run_id, SecureRandom.uuid)
+        test_dir = params.fetch(:results_output_dir, File.join(Dir.tmpdir(), run_uuid))
+
+        # Set up the log file and output directory
+        FileUtils.mkdir_p(test_dir)
+        Fastlane::Actions.lane_context[:FIREBASE_TEST_LOG_FILE_PATH] = File.join(test_dir, 'output.log')
+
+        device = Fastlane::FirebaseDevice.new(
+          model: params[:model],
+          version: params[:version],
+          locale: params[:locale],
+          orientation: params[:orientation]
+        )
+
+        result = test_runner.run_tests(
+          apk_path: params[:apk_path],
+          test_apk_path: params[:test_apk_path],
+          device: device,
+          type: params[:type]
+        )
+
+        # Download all of the outputs from the job to the local machine
+        test_runner.download_result_files(
+          result: result,
+          destination: test_dir,
+          project_id: params[:project_id],
+          key_file_path: params[:key_file]
+        )
+
+        FastlaneCore::UI.test_failure! "Firebase Tests failed – more information can be found at #{result.more_details_url}" unless result.success?
+
+        UI.success 'Firebase Tests Complete'
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        'Runs the specified tests in Firebase Test Lab'
+      end
+
+      def self.details
+        description
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(
+            key: :project_id,
+            env_name: 'GCP_PROJECT',
+            description: 'The Project ID to test in',
+            type: String
+          ),
+          FastlaneCore::ConfigItem.new(
+            key: :key_file,
+            env_name: 'GOOGLE_APPLICATION_CREDENTIALS',
```

Comment:
Makes sense – I'm not sure this one needs an env var since this is always provided by some sort of secrets implementation anyway.

Removed in 08bc2c07913c28982236edf431d803e52db33723.